firstvoices/backend/tests/test_apis/base_api_test.py

```diff
@@ -378,6 +417,52 @@ def test_detail_403_entry_not_visible(self):
 
         assert response.status_code == 403
 
+    @pytest.mark.django_db
+    def test_update_assistant_permissions_valid(self):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.ASSISTANT
+        )
+
+        instance = self.create_minimal_instance(site=site, visibility=Visibility.TEAM)
+
+        self.client.force_authenticate(user=user)
+
+        data = self.get_valid_data(site)
+        data["visibility"] = "team"
+
+        response = self.client.put(
+            self.get_detail_endpoint(key=instance.slug, site_slug=site.slug)
+            if instance.__class__ == SitePage
+            else self.get_detail_endpoint(key=instance.id, site_slug=site.slug),
+            data=self.format_upload_data(data),
+            content_type=self.content_type,
+        )
+
+        assert response.status_code == 200
+
+    @pytest.mark.django_db
+    def test_update_assistant_permissions_invalid(self):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.ASSISTANT
+        )
+
+        instance = self.create_minimal_instance(site=site, visibility=Visibility.TEAM)
+
+        self.client.force_authenticate(user=user)
+
+        data = self.get_valid_data(site)
+        data["visibility"] = "public"
+
+        response = self.client.put(
+            self.get_detail_endpoint(key=instance.slug, site_slug=site.slug)
+            if instance.__class__ == SitePage
+            else self.get_detail_endpoint(key=instance.id, site_slug=site.slug),
+            data=self.format_upload_data(data),
+            content_type=self.content_type,
+        )
+
+        assert response.status_code == 400
```

Comment:
Same comment-- these are great tests but please put them in a mixin for update APIs

Reply:
Done 😄